### PR TITLE
Signup and update account

### PR DESCRIPTION
**Pull request by nhs3108**

### CODE REVIEW

File: app/views/users/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        <%= f.password_field :password, class: "form-control" %>
+      </div>
+      <div class="field">
+        <%= f.label :password_confirmation %><br>
+        <%= f.password_field :password_confirmation, class: "form-control" %>
+      </div>
+      <br>
+        <%= f.file_field :avatar, class: "filestyle", id: "user_avatar", accept: Settings.user.avatar.accepted_format %>
+      <br>
+        <%= f.submit t("done"), class: "btn btn-primary pull-right" %>
```
Comment (phamkykhoi) [inline app/views/users/new.html.erb]: Indent

File: app/models/user.rb [hunk 2]
```diff
@@ -30,22 +39,67 @@ def User.new_token
...
+  def authenticated?(attribute, token)
+    digest = send("#{attribute}_digest")
+    return false if digest.nil?
+    BCrypt::Password.new(digest).is_password? token
+  end
+
   def remember
     self.remember_token = User.new_token
-    update_attribute(:remember_digest, User.digest(remember_token))
+    update_attribute :remember_digest, User.digest(remember_token)
```
Comment (dieunb) [inline app/models/user.rb]: use `update_attributes` instead of `update_attribute`

File: app/models/user.rb [hunk 3]
```diff
@@ -30,22 +39,67 @@ def User.new_token
...
   end
 
   def forget
-    update_attribute(:remember_digest, nil)
+    update_attribute :remember_token, nil
```
Comment (dieunb) [inline app/models/user.rb]: use `update_attributes` instead of `update_attribute`

File: app/models/user.rb [hunk 4]
```diff
@@ -30,22 +39,67 @@ def User.new_token
...
   end
-  
+
+  def following? other_user
+    following.include? other_user
+  end
+
+  def activate
+    update_attributes :activated, true
+    update_attributes :activated_at, Time.zone.now
```
Comment (dieunb) [inline app/models/user.rb]: Merged 2 lines to one line

File: app/models/user.rb [hunk 5]
```diff
@@ -30,22 +39,67 @@ def User.new_token
...
+  end
+
+  def send_password_reset_email
+    UserMailer.password_reset(self).deliver_now
+  end
+
+  def create_reset_digest
+    self.reset_token = User.new_token
+    update_attributes :reset_digest, User.digest(reset_token)
+    update_attributes :reset_at, Time.zone.now
```
Comment (dieunb) [inline app/models/user.rb]: merge 2 lines code to one line

File: app/models/user.rb [hunk 6]
```diff
@@ -30,22 +39,67 @@ def User.new_token
...
   private
   def downcase_email
     self.email = email.downcase
   end
-end
\ No newline at end of file
+
+  def avatar_size
+    if avatar.size > Settings.user.avatar.max_size_in_MB.megabytes
+      errors.add :avatar, I18n.t("should_be_less_than") + " " +
```
Comment (dieunb) [inline app/models/user.rb]: use i18n variable instead it

**Pull request merged**